src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c

```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjBlockstop_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809466F0.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].first;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+    while (tempActor) {
```

Comment:
We compare pointers to NULL where possible
```suggestion
    while (tempActor != NULL) {
```